pubsub/subscriptions/main.go

```diff
@@ -0,0 +1,150 @@
+	client, err := pubsub.NewClient(ctx, proj)
+	if err != nil {
+		log.Fatalf("Could not create pubsub Client: %v", err)
+	}
+	// [END auth]
+
+	// Print all the subscriptions in the project.
+	fmt.Println("Listing all subscriptions from the project:")
+	subs, err := list(client)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, sub := range subs {
+		fmt.Printf("%v\n", sub.Name())
+	}
+
+	const topic = "example-topic"
+	// Create a topic to subscribe to.
+	t, err := client.NewTopic(ctx, topic)
+	if err != nil {
+		log.Fatalf("Failed to create the topic: %v", err)
+	}
+	defer t.Delete(ctx) // cleanup when finished using t.
+
+	const sub = "example-subscription"
+	// Create a new subscription.
+	if err := create(client, sub, t); err != nil {
+		log.Fatal(err)
+	}
+
+	// Pull messages via the subscription.
+	if err := pullMsgs(client, sub, t); err != nil {
+		log.Fatal(err)
+	}
+
+	// Delete the subscription.
+	if err := delete(client, sub); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func list(c *pubsub.Client) ([]*pubsub.Subscription, error) {
+	ctx := context.Background()
+	// [START get_all_subscriptions]
+	var subs []*pubsub.Subscription
+	it := c.Subscriptions(ctx)
+	for {
+		s, err := it.Next()
+		if err == pubsub.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Failed to list subscriptions: %v", err)
+		}
+		subs = append(subs, s)
+	}
+	// [END get_all_subscriptions]
+	return subs, nil
+}
+
+func pullMsgs(c *pubsub.Client, name string, topic *pubsub.Topic) error {
+	ctx := context.Background()
+
+	const n = 10
+	// publish 10 messages on the topic.
+	for i := 0; i < n; i++ {
+		_, err := topic.Publish(ctx, &pubsub.Message{
+			Data: []byte(fmt.Sprintf("hello world #%d", i)),
+		})
+		if err != nil {
+			return fmt.Errorf("Failed to publish message #%d: %v", i, err)
+		}
+	}
+
+	// [START pull_messages]
+	sub := c.Subscription(name)
+	it, err := sub.Pull(ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to pull from subsription: %v", err)
+	}
+	defer it.Stop()
+
+	// Consume 10 messages.
+	for i := 0; i < n; i++ {
+		msg, err := it.Next()
+		if err == pubsub.Done {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("Failed when iterating on messages: %v", err)
+		}
+		fmt.Printf("Got message: %q\n", string(msg.Data))
+		msg.Done(true)
+	}
+	// [END pull_messages]
+	return nil
+}
+
+func create(c *pubsub.Client, name string, topic *pubsub.Topic) error {
+	ctx := context.Background()
+	// [START create_subscription]
+	sub, err := c.NewSubscription(ctx, name, topic, 10*time.Second, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to create a new subscription: %v", err)
+	}
+	fmt.Printf("Created subscription: %v\n", sub)
+	// [END create_subscription]
+	return nil
+}
+
+func delete(c *pubsub.Client, name string) error {
+	ctx := context.Background()
+	// [START delete_subscription]
+	sub := c.Subscription(name)
+	if err := sub.Delete(ctx); err != nil {
+		return fmt.Errorf("Failed to delete the subscription: %v\n", err)
```

Comment:
I don't think we need to wrap the errors for the samples. Also, don't put newlines at the end of wrapper errors.